plugins/module_utils/redfish_utils.py

```diff
@@ -3544,3 +3545,111 @@ def delete_volumes(self, storage_subsystem_id, volume_ids):
+        self.storage_subsystems_uris = [i['@odata.id'] for i in response['data'].get('Members', [])]
+        if not self.storage_subsystems_uris:
+            return {
+                'ret': False,
+                'msg': "StorageCollection's Members array is either empty or missing"}
+
+        # Matching Storage Subsystem ID with user input
+        self.storage_subsystem_uri = ""
+        for storage_subsystem_uri in self.storage_subsystems_uris:
+            if storage_subsystem_uri.split("/")[-2] == storage_subsystem_id:
+                self.storage_subsystem_uri = storage_subsystem_uri
+
+        if not self.storage_subsystem_uri:
+            return {
+                'ret': False,
+                'msg': "Provided Storage Subsystem ID %s does not exist on the server" % storage_subsystem_id}
+
+        # Validate input parameters
+        required_parameters = ['RAIDType', 'Drives', 'CapacityBytes']
+        allowed_parameters = ['DisplayName', 'InitializeMethod', 'MediaSpanCount',
+                              'Name', 'ReadCachePolicy', 'StripSizeBytes', 'VolumeUsage', 'WriteCachePolicy']
+
+        for parameter in required_parameters:
+            if not volume_details.get(parameter):
+                return {
+                    'ret': False,
+                    'msg': "%s are required parameter to create a volume" % str(required_parameters)}
+
+        # Navigate to the volume uri of the correct storage subsystem
+        response = self.get_request(self.root_uri + self.storage_subsystem_uri)
+        if response['ret'] is False:
+            return response
+        data = response['data']
+
+        # Deleting any volumes of RAIDType None present on the Storage Subsystem
+        response = self.get_request(self.root_uri + data['Volumes']['@odata.id'])
+        if response['ret'] is False:
+            return response
+        volume_data = response['data']
+
+        if "Members" in volume_data:
+            for member in volume_data["Members"]:
+                response = self.get_request(self.root_uri + member['@odata.id'])
+                if response['ret'] is False:
+                    return response
+                member_data = response['data']
+
+                if member_data["RAIDType"] == "None":
+                    response = self.delete_request(self.root_uri + member['@odata.id'])
+                    if response['ret'] is False:
+                        return response
+
+        # Construct payload and issue POST command to create volume
+        volume_details["Links"] = {}
+        volume_details["Links"]["Drives"] = []
+        for drive in volume_details["Drives"]:
+            volume_details["Links"]["Drives"].append({"@odata.id": drive})
+        del volume_details["Drives"]
+        payload = volume_details
+        response = self.post_request(self.root_uri + data['Volumes']['@odata.id'], payload)
+        if response['ret'] is False:
+            # Fail message if run for iLO5 or lower server
```

Comment:
I still don't think this is really in our best interest to add... We don't perform additional version checks in failure paths anywhere else, and this seems like it'll be nearly impossible to maintain across the board.

Reply:
Actually, this version specific code only comes into picture for iLO6 RAID/Volume related operations. All other functionalities that we maintain do not have any version dependencies.

Reply:
That's not the issue; the issue is establishing a pattern of:

```
if FAIL:
    if vendor == "X":
        # Add version checks and modify error message for vendor X
    elif vendor == "Y":
        # Add version checks and modify error message for vendor Y
    elif vendor == "Z":
        # Add version checks and modify error message for vendor Z
    elif ...
```

I don't think this pattern scales well or is very maintainable. The service should already be returning the proper information to convey that it doesn't support the given request, such as a "method not allowed" message, which is a clear indication the service does not support POST to the VolumeCollection.

Reply:
Hmm, will defer to your expertise on the matter. Updated the code to remove this.